Lib/idlelib/idle_test/test_format.py

```diff
@@ -611,47 +611,57 @@ def test_change_indentwidth(self, askinteger):
 
 class RstripTest(unittest.TestCase):
 
-    def test_rstrip_line(self):
-        editor = MockEditor()
-        text = editor.text
-        do_rstrip = ft.Rstrip(editor).do_rstrip
-        eq = self.assertEqual
+    @classmethod
+    def setUpClass(cls):
+        requires('gui')
+        cls.root = Tk()
+        cls.root.withdraw()
+        cls.text = Text(cls.root)
+        cls.editor = MockEditor(text=cls.text)
+        cls.do_rstrip = ft.Rstrip(cls.editor).do_rstrip
 
-        do_rstrip()
-        eq(text.get('1.0', 'insert'), '')
-        text.insert('1.0', '     ')
-        do_rstrip()
-        eq(text.get('1.0', 'insert'), '')
-        text.insert('1.0', '     \n')
-        do_rstrip()
-        eq(text.get('1.0', 'insert'), '\n')
-
-    def test_rstrip_multiple(self):
-        editor = MockEditor()
-        #  Comment above, uncomment 3 below to test with real Editor & Text.
-        #from idlelib.editor import EditorWindow as Editor
-        #from tkinter import Tk
-        #editor = Editor(root=Tk())
-        text = editor.text
-        do_rstrip = ft.Rstrip(editor).do_rstrip
+    @classmethod
+    def tearDownClass(cls):
+        del cls.text, cls.do_rstrip, cls.editor
+        cls.root.update_idletasks()
+        cls.root.destroy()
+        del cls.root
+
+    def tearDown(self):
+        self.text.delete('1.0', 'end')
```

Comment:
Shouldn't this be `self.text.delete('1.0', 'end-1c')`?

Reply:
For 'convenience', I guess, 'end' == 'end-1c' for delete and insert, though not for get and index.  So this is a standard idiom for 'clear'.   I have found this confusing and it was a nuisance for writing mock Text, and will be a nuisance for implementing 'end-nc' for the mock, as needed now for RstripTest (and probably some others).  This is why I instead switched to the real Text, at least for now.

I made the change to be explicit that we are not trying to delete tk's undeletable but not invisible guard newline.